.github/workflows/push_modules_to_subrepo.yml

```
@@ -0,0 +1,49 @@
+name: Push modules to subrepo
+
+on:
+  push:
+    branches:
+      - main
+    paths:
+      - "infra/modules/**"
+
+concurrency:
+  group: ${{ github.workflow }}
+  cancel-in-progress: false
+      
+jobs:
+  get-modified-modules:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout repository
+        uses: actions/checkout@v4
+        with:
+            fetch-depth: 2 # Ensure we have enough history for the diff
+
+      - name: Get modified modules
+        id: get-modules
+        run: |
+          folders_list=$(git diff --no-commit-id --name-only HEAD~1 infra/modules | xargs -r dirname | sed 's#^infra\/modules/\([^/]*\).*#\1#' | uniq | jq -R -c . | jq -s -c .)
```

what about something like...

```suggestion
folders_list=$(git diff --name-only ${{ github.event.before }} ${{ github.sha }} infra/modules | ...)
```